### PR TITLE
Mule 10722: Not Handling FTP Connection Errors in FTPMessageReceiver

### PR DESCRIPTION
**Pull request by mbuchwald**

### CODE REVIEW

File: transports/ftp/src/test/java/org/mule/transport/ftp/FtpReconnectionTestCase.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+import org.junit.Test;
+import org.junit.runners.Parameterized;
+
+/**
+ * Verify that some Connect Exceptions-like are being handled and result in reconnections.
+ */
+public class FtpReconnectionTestCase extends AbstractFtpServerTestCase
+{
+    protected TestSystemExceptionStrategy tryReconnectionStrategy;
+    MockFailingServer server;
```
Comment (pablolagreca) [inline transports/ftp/src/test/java/org/mule/transport/ftp/FtpReconnectionTestCase.java]: make it private

File: transports/ftp/src/test/java/org/mule/transport/ftp/FtpReconnectionTestCase.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runners.Parameterized;
+
+/**
+ * Verify that some Connect Exceptions-like are being handled and result in reconnections.
+ */
+public class FtpReconnectionTestCase extends AbstractFtpServerTestCase
+{
+    protected TestSystemExceptionStrategy tryReconnectionStrategy;
```
Comment (pablolagreca) [inline transports/ftp/src/test/java/org/mule/transport/ftp/FtpReconnectionTestCase.java]: private

File: transports/ftp/src/test/java/org/mule/transport/ftp/FtpReconnectionTestCase.java [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+            createFileOnFtpServer("lostConnection/test1");
+            FlowExecutionListener flowExecutionListener = new FlowExecutionListener("LostConnection", muleContext);
+            flowExecutionListener.waitUntilFlowIsComplete();
+        }
+        catch (java.lang.AssertionError e)
+        {
+            assertTrue(tryReconnectionStrategy.reconnect);
+            return;
+        }
+        fail("Should try to reconnect");
```
Comment (pablolagreca) [inline transports/ftp/src/test/java/org/mule/transport/ftp/FtpReconnectionTestCase.java]: move fail(..) after flowExecutionListener.waitUntilFlowIsComplete()

File: transports/ftp/src/test/java/org/mule/transport/ftp/FtpReconnectionTestCase.java (hunk 2 above)
Comment (mbuchwald) [inline transports/ftp/src/test/java/org/mule/transport/ftp/FtpReconnectionTestCase.java]: fixed

File: transports/ftp/src/test/java/org/mule/transport/ftp/FtpReconnectionTestCase.java (hunk 1 above)
Comment (mbuchwald) [inline transports/ftp/src/test/java/org/mule/transport/ftp/FtpReconnectionTestCase.java]: fixed

File: transports/ftp/src/test/java/org/mule/transport/ftp/FtpReconnectionTestCase.java (hunk 3 above)
Comment (mbuchwald) [inline transports/ftp/src/test/java/org/mule/transport/ftp/FtpReconnectionTestCase.java]: Fixed the way it was being tested (using PollingProber instead)

**Pull request merged**